sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -1425,6 +1427,7 @@ def _snapshot_to_json(snapshot: Snapshot) -> str:
 
 def parse_snapshot(
     model_cache: ModelCache,
+    optimized_query_cache: OptimizedQueryCache,
```

Comment:
Let's not do this here. I'm going to work on a more comprehensive snapshot caching next.